### ISSUE TITLE
No platform compatability analyzer warnings in Razor @code sections

### ISSUE THREAD

**Issue by lambdageek** [dotnet/razor-compiler#216]

**Comment by lambdageek** [comment]
Related to https://github.com/dotnet/roslyn/issues/39758

**Comment by lambdageek** [comment]
This is also relevant to MAUI Blazor projects if a MAUI component calls a platform-specific API that is not available on a particular platform.

**Comment by marek-safar** [comment]
This is also very likely problematic for trimming warning analyzer

/cc @agocke

**Comment by agocke** [comment]
Unless there's something specific to this analyzer, this sounds like a Roslyn bug. It might be marked as generated code, which will silence analyzer warnings by default, but in this case I think it should probably classify as user code.

cc @chsienki

**Comment by chsienki** [comment]
By default generator created code is marked as user code. 

However, the generator author can choose to give a hint name ending in `.g.cs` or add `/// <Autogenerated>` to the top of the file. That will cause the regular analyzer heuristics to kick in and treat it as generated code, much the same as if a tool had added it outside the compilation pipeline.

**Comment by agocke** [comment]
@chsienki This bug is that warnings *don't* appear in razor code sections, though. Are either of those things added to razor compiled files?

**Comment by chsienki** [comment]
@agocke yes. https://github.com/dotnet/sdk/blob/4e35c47e93f5ac1b8b4d813fd4b3f69e2377afa7/src/RazorSdk/SourceGenerators/RazorSourceGenerator.cs#L240

**Comment by agocke** [comment]
Got it, so then this looks like a Razor bug to me. If there's scaffolding code for Razor, that might be considered generated, but the stuff in `@code` looks like user-written code to me and shouldn't be marked as generated.
